### ISSUE TITLE
Multiple listen on https, works only one, last

### ISSUE THREAD

**Issue by webhtb** [h2o/h2o#717]
Hi, i try to use multiple listen in one hosts and i discover only the h2o respond only in last ip.
Also i checked on netstat, the h2o bind in all ips but respond to browsers only in last ip, it is about any https host

```
  "host.com:443":
    listen:
      host: 192.168.0.101
      port: 443
    listen:
      host: 127.0.0.1
      port: 443
      ssl:
        certificate-file: /var/www/ssl/cert.crt
        key-file: /var/www/ssl/key.key
    paths:
      "/":
........
```

**Comment by kazuho** [comment]
Is my assumption correct that you want to server `HTTPS` (not `HTTP`) on `192.168.0.101:443`?

If that is the case, you should define `ssl` attribute to the listen directive for `192.168.0.101:443`.

**Comment by webhtb** [comment]
It is listen, also no matter the ip, i discover this with real domains and real ips, this is only to reproduce local and i got same result. I told you, the h2o make bind on all ips but respond to browsers only for last ip in the list. Why multiple? Because one ipv4 and one ipv6 for each vhost.

**Comment by kazuho** [comment]
> h2o make bind on all ips but respond to browsers only for last ip in the list.

Doesn't the server respond to the browser if you access the address using __HTTP__ (not HTTPS)?

What I am saying is that you should write something like:
```
  "host.com:443":
    listen:
      host: 192.168.0.101
      port: 443
      ssl:                                       # this attribute is required to have this listen accept HTTPS connections
        certificate-file: /var/www/ssl/cert.crt  # 
        key-file: /var/www/ssl/key.key           #
    listen:
      host: 127.0.0.1
      port: 443
      ssl:
        certificate-file: /var/www/ssl/cert.crt
        key-file: /var/www/ssl/key.key
    paths:
      "/":
........
```

**Comment by webhtb** [comment]
i changed the conf like you told me and see this:

[root@192-168-0-101 h2o]# /usr/local/bin/h2o -c /usr/local/etc/h2o.conf 
*** Error in `/usr/local/bin/h2o': double free or corruption (!prev): 0x00000000026c1130 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x77da5)[0x7f02e012dda5]
/lib64/libc.so.6(+0x804fa)[0x7f02e01364fa]
/lib64/libc.so.6(cfree+0x4c)[0x7f02e0139cac]
/usr/local/bin/h2o(h2o_config_dispose_pathconf+0x33)[0x455b13]
/usr/local/bin/h2o[0x464b38]
/usr/local/bin/h2o(h2o_mimemap_define_dynamic+0x8f)[0x46604f]
/usr/local/bin/h2o[0x456ea8]
/usr/local/bin/h2o(h2o_configurator_apply_commands+0x377)[0x456b57]
/usr/local/bin/h2o[0x457a30]
/usr/local/bin/h2o(h2o_configurator_apply_commands+0x3f7)[0x456bd7]
/usr/local/bin/h2o(h2o_configurator_apply+0x3a)[0x457bba]
/usr/local/bin/h2o(main+0x4f2)[0x449ac2]
/lib64/libc.so.6(__libc_start_main+0xf0)[0x7f02e00d6580]
/usr/local/bin/h2o(_start+0x29)[0x44a169]
======= Memory map: ========
00400000-0073f000 r-xp 00000000 08:01 4198428                            /usr/local/bin/h2o
0093e000-0093f000 r--p 0033e000 08:01 4198428                            /usr/local/bin/h2o
0093f000-00950000 rw-p 0033f000 08:01 4198428                            /usr/local/bin/h2o
00950000-00953000 rw-p 00000000 00:00 0 
02684000-026c6000 rw-p 00000000 00:00 0                                  [heap]
7f02d0000000-7f02d0021000 rw-p 00000000 00:00 0 
7f02d0021000-7f02d4000000 ---p 00000000 00:00 0 
7f02d4000000-7f02d4021000 rw-p 00000000 00:00 0 
7f02d4021000-7f02d8000000 ---p 00000000 00:00 0 
7f02d8000000-7f02d8021000 rw-p 00000000 00:00 0 
7f02d8021000-7f02dc000000 ---p 00000000 00:00 0 
7f02dec8b000-7f02deca1000 r-xp 00000000 08:01 3939554                    /usr/lib64/libgcc_s-5.3.1-20151207.so.1
7f02deca1000-7f02deea0000 ---p 00016000 08:01 3939554                    /usr/lib64/libgcc_s-5.3.1-20151207.so.1
7f02deea0000-7f02deea1000 r--p 00015000 08:01 3939554                    /usr/lib64/libgcc_s-5.3.1-20151207.so.1
7f02deea1000-7f02deea2000 rw-p 00016000 08:01 3939554                    /usr/lib64/libgcc_s-5.3.1-20151207.so.1
7f02deea2000-7f02deea3000 ---p 00000000 00:00 0 
7f02deea3000-7f02df6a3000 rw-p 00000000 00:00 0                          [stack:30152]
7f02df6a3000-7f02df6a4000 ---p 00000000 00:00 0 
7f02df6a4000-7f02dfea4000 rw-p 00000000 00:00 0                          [stack:30150]
7f02dfea4000-7f02dfeaf000 r-xp 00000000 08:01 3938612                    /usr/lib64/libnss_files-2.22.so
7f02dfeaf000-7f02e00ae000 ---p 0000b000 08:01 3938612                    /usr/lib64/libnss_files-2.22.so
7f02e00ae000-7f02e00af000 r--p 0000a000 08:01 3938612                    /usr/lib64/libnss_files-2.22.so
7f02e00af000-7f02e00b0000 rw-p 0000b000 08:01 3938612                    /usr/lib64/libnss_files-2.22.so
7f02e00b0000-7f02e00b6000 rw-p 00000000 00:00 0 
7f02e00b6000-7f02e026d000 r-xp 00000000 08:01 3938592                    /usr/lib64/libc-2.22.so
7f02e026d000-7f02e046d000 ---p 001b7000 08:01 3938592                    /usr/lib64/libc-2.22.so
7f02e046d000-7f02e0471000 r--p 001b7000 08:01 3938592                    /usr/lib64/libc-2.22.so
7f02e0471000-7f02e0473000 rw-p 001bb000 08:01 3938592                    /usr/lib64/libc-2.22.so
7f02e0473000-7f02e0477000 rw-p 00000000 00:00 0 
7f02e0477000-7f02e047e000 r-xp 00000000 08:01 3938624                    /usr/lib64/librt-2.22.so
7f02e047e000-7f02e067d000 ---p 00007000 08:01 3938624                    /usr/lib64/librt-2.22.so
7f02e067d000-7f02e067e000 r--p 00006000 08:01 3938624                    /usr/lib64/librt-2.22.so
7f02e067e000-7f02e067f000 rw-p 00007000 08:01 3938624                    /usr/lib64/librt-2.22.so
7f02e067f000-7f02e0682000 r-xp 00000000 08:01 3938598                    /usr/lib64/libdl-2.22.so
7f02e0682000-7f02e0881000 ---p 00003000 08:01 3938598                    /usr/lib64/libdl-2.22.so
7f02e0881000-7f02e0882000 r--p 00002000 08:01 3938598                    /usr/lib64/libdl-2.22.so
7f02e0882000-7f02e0883000 rw-p 00003000 08:01 3938598                    /usr/lib64/libdl-2.22.so
7f02e0883000-7f02e089b000 r-xp 00000000 08:01 3938620                    /usr/lib64/libpthread-2.22.so
7f02e089b000-7f02e0a9a000 ---p 00018000 08:01 3938620                    /usr/lib64/libpthread-2.22.so
7f02e0a9a000-7f02e0a9b000 r--p 00017000 08:01 3938620                    /usr/lib64/libpthread-2.22.so
7f02e0a9b000-7f02e0a9c000 rw-p 00018000 08:01 3938620                    /usr/lib64/libpthread-2.22.so
7f02e0a9c000-7f02e0aa0000 rw-p 00000000 00:00 0 
7f02e0aa0000-7f02e0ab5000 r-xp 00000000 08:01 3940058                    /usr/lib64/libz.so.1.2.8
7f02e0ab5000-7f02e0cb4000 ---p 00015000 08:01 3940058                    /usr/lib64/libz.so.1.2.8
7f02e0cb4000-7f02e0cb5000 r--p 00014000 08:01 3940058                    /usr/lib64/libz.so.1.2.8
7f02e0cb5000-7f02e0cb6000 rw-p 00015000 08:01 3940058                    /usr/lib64/libz.so.1.2.8
7f02e0cb6000-7f02e0db7000 r-xp 00000000 08:01 3938600                    /usr/lib64/libm-2.22.so
7f02e0db7000-7f02e0fb6000 ---p 00101000 08:01 3938600                    /usr/lib64/libm-2.22.so
7f02e0fb6000-7f02e0fb7000 r--p 00100000 08:01 3938600                    /usr/lib64/libm-2.22.so
7f02e0fb7000-7f02e0fb8000 rw-p 00101000 08:01 3938600                    /usr/lib64/libm-2.22.so
7f02e0fb8000-7f02e0fd9000 r-xp 00000000 08:01 3933826                    /usr/lib64/ld-2.22.so
7f02e11a9000-7f02e11ae000 rw-p 00000000 00:00 0 
7f02e11d4000-7f02e11d8000 rw-p 00000000 00:00 0 
7f02e11d8000-7f02e11d9000 r--p 00020000 08:01 3933826                    /usr/lib64/ld-2.22.so
7f02e11d9000-7f02e11da000 rw-p 00021000 08:01 3933826                    /usr/lib64/ld-2.22.so
7f02e11da000-7f02e11db000 rw-p 00000000 00:00 0 
7fff2ea2b000-7fff2ea4c000 rw-p 00000000 00:00 0                          [stack]
7fff2eb46000-7fff2eb48000 r--p 00000000 00:00 0                          [vvar]
7fff2eb48000-7fff2eb4a000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
fetch-ocsp-response (using OpenSSL 1.0.2e-fips 3 Dec 2015)
fetch-ocsp-response (using OpenSSL 1.0.2e-fips 3 Dec 2015)
sending OCSP request to http://ocsp.int-x1.letsencrypt.org/
sending OCSP request to http://ocsp.int-x1.letsencrypt.org/
Aborted (core dumped)
[root@192-168-0-101 h2o]# /var/www/ssl/cert.crt: good
	This Update: Jan 20 10:00:00 2016 GMT
	Next Update: Jan 27 10:00:00 2016 GMT
verifying the response signature
/var/www/ssl/cert.crt: good
	This Update: Jan 20 10:00:00 2016 GMT
	Next Update: Jan 27 10:00:00 2016 GMT
verifying the response signature
verify OK (used: -VAfile /tmp/OQLDv8N0Jp/issuer.crt)
verify OK (used: -VAfile /tmp/WFxAvfSYV_/issuer.crt)

**Comment by kazuho** [comment]
Can you please provide (or create) the configuration file that causes the crash?

**Comment by webhtb** [comment]
[h2o.conf.txt](https://github.com/h2o/h2o/files/100462/h2o.conf.txt)

**Comment by webhtb** [comment]
Please make h2o like beta3 or solve this, in beta3 we just repeat the host and all was OK, now in beta 4 we can`t, h2o now say duplicate host, that was the only way to make this work with more than one ip/host:

  "host.com:443":
    listen:
      host: 192.168.0.101
      port: 443


  "host.com:443":
    listen:
      host: 127.0.0.1
      port: 443

**Comment by kazuho** [comment]
Thank you for the configuration file.

I believe that the crash has been fixed with the merger of #723.  Can you check if the issue persists?

**Comment by webhtb** [comment]
Now in my computer is OK, will see tonight in servers! Thanks!!!
